src/services/storage/handlers/offer.ts

```diff
@@ -46,26 +47,28 @@ const handler: Handler<StorageServices> = {
           break
         }
 
-        const [firstMsg, ...restMsg] = msg[0].replace('0x', '')
-        const flag = firstMsg.substring(0, 2)
+        const [firstMsg, ...restMsg] = msg
+        const flag = firstMsg.substring(2, 4)
 
         if (flag === '01') { // PeerId definition
-          offer.peerId = decodeByteArray([firstMsg.substring(2), ...restMsg])
+          offer.peerId = decodeByteArray([`0x${firstMsg.substring(4)}`, ...restMsg])
```

Comment:
Hmm you don't need to reconstruct the `0x` of the message. See how `hexToAscii` is implemented [here](https://github.com/ethereum/web3.js/blob/1.x/packages/web3-utils/src/index.js#L168).
But since you have it here, you can already keep it I guess :-)